tests/arch/x86/pagetables/src/main.c

```diff
@@ -233,7 +233,11 @@ void z_vrfy_dump_my_ptables(void)
  *
  * @ingroup kernel_memprotect_tests
  */
-void test_dump_ptables(void)
+#ifdef CONFIG_USERSPACE
+ZTEST_USER(x86_pagetables, test_dump_ptables)
+#else
+ZTEST(x86_pagetables, test_dump_ptables)
+#endif
```

Comment:
`ZTEST_USER` will default to `ZTEST` if `CONFIG_USERSPACE=n`, you don't need this, can just use `ZTEST_USER`

Reply:
Thanks for correcting this. Fixed.
May I consult a question with you? If I want to run test_dump_ptables() twice, one is on userspace and another is not,  is there any way I can do it? Thanks! According to the original tests:
```
        ztest_unit_test(test_dump_ptables),
        ztest_user_unit_test(test_dump_ptables)
```

Reply:
Yes, you could do something like:

```
static void test_dump_ptables_common(void)
{
  // Test logic here
}

ZTEST_USER(x86_pagetables, test_dump_ptables_user)
{
  test_dump_ptables_common();
}

ZTEST(x86_pagetables, test_dump_ptables)
{
  test_dump_ptables_common();
}
```

Reply:
We should have this resolved before merging, as it stands this test will only run once in usermode unlike the original logic.

Reply:
Fixed. Thanks for your example!